### PR TITLE
fix(Mesh): recompute adjacency only for triangulated elements

### PR DESCRIPTION
**Pull request by panquez**

### CODE REVIEW

File: src/geode/mesh/helpers/convert_surface_mesh.cpp [hunk 1]
```diff
@@ -346,23 +379,31 @@ namespace geode
         SurfaceMeshBuilder< dimension >& builder )
     {
         std::vector< bool > to_delete( surface.nb_polygons(), false );
-        for( const auto p : geode::Range{ surface.nb_polygons() } )
+        for( const auto p : Range{ surface.nb_polygons() } )
         {
             const auto nb_vertices = surface.nb_polygon_vertices( p );
             to_delete[p] = nb_vertices != 3;
             if( nb_vertices > 3 )
             {
```
Comment (MelchiorSchuh) [inline src/geode/mesh/helpers/convert_surface_mesh.cpp:387]:
```suggestion
            to_delete[p] = nb_vertices != 3;
            if( to_delete[p] )
            {
```

File: src/geode/mesh/helpers/convert_surface_mesh.cpp [hunk 2]
```diff
@@ -271,6 +271,39 @@ namespace
...
+        absl::Span< const geode::index_t > new_polygons )
+    {
+        if( adjacents.front() )
+        {
+            builder.set_polygon_adjacent(
+                { new_polygons.front(), 0 }, adjacents.front()->polygon_id );
+            builder.set_polygon_adjacent(
+                adjacents.front().value(), new_polygons.front() );
+        }
+        for( const auto v : LRange{ 1, nb_vertices - 1 } )
```
Comment (MelchiorSchuh) [inline src/geode/mesh/helpers/convert_surface_mesh.cpp:288]: nb_vertices is not defined here

**Review by MelchiorSchuh — APPROVED** [review]

File: src/geode/mesh/helpers/convert_surface_mesh.cpp [hunk 3]
```diff
@@ -346,23 +379,31 @@ namespace geode
...
             to_delete[p] = nb_vertices != 3;
             if( nb_vertices > 3 )
             {
+                absl::FixedArray< std::optional< PolygonEdge > > adjacents(
+                    nb_vertices, std::nullopt );
+                for( const auto e : LRange{ nb_vertices } )
+                {
+                    adjacents[e] = surface.polygon_adjacent_edge( { p, e } );
+                }
+                std::vector< index_t > new_polygons;
```
Comment (BotellaA) [inline src/geode/mesh/helpers/convert_surface_mesh.cpp:394]: Could also be a FixedArray

File: src/geode/mesh/helpers/convert_surface_mesh.cpp [hunk 4]
```diff
@@ -271,6 +271,39 @@ namespace
...
+            }
+        }
+        if( adjacents.back() )
+        {
+            builder.set_polygon_adjacent(
+                { new_polygons.back(), 2 }, adjacents.back()->polygon_id );
+            builder.set_polygon_adjacent(
+                adjacents.back().value(), new_polygons.back() );
+        }
+        builder.compute_polygon_adjacencies( new_polygons );
```
Comment (BotellaA) [inline src/geode/mesh/helpers/convert_surface_mesh.cpp:305]: Why not, a bit more computation but it works

**Review by BotellaA — APPROVED** [review]

**Pull request merged**